File: backend/app/schemas/lesson.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LessonRead(BaseModel):
    id: int
    class_subject_requirement_id: int
    sequence_number: int
    fixed_time_slot_id: int | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class LessonFixTimeSlot(BaseModel):
    """Body for PATCH /lessons/{id}/fix-time-slot. time_slot_id is required
    (not defaulted) so the caller must explicitly say null to clear a fixed
    slot, rather than omitting the field by accident."""

    time_slot_id: int | None

File: frontend/src/types/lesson.ts
// Mirrors backend/app/schemas/lesson.py::LessonRead in full, including
// Task 28's fixed_time_slot_id. Lesson has no create/update/remove
// endpoints on the backend (rows are only produced by generate-lessons),
// so there are no corresponding Create/Update interfaces here -- unlike
// every other resource module, a full 5-set would describe endpoints that
// don't exist.

export interface Lesson {
  id: number;
  class_subject_requirement_id: number;
  sequence_number: number;
  fixed_time_slot_id: number | null;
  created_at: string;
  updated_at: string;
}

// Body for PATCH /lessons/{id}/fix-time-slot (Task 28).
export interface LessonFixTimeSlot {
  time_slot_id: number | null;
}
